### ISSUE TITLE
Problem with emoji characters on mac

### ISSUE THREAD

**Issue by Cod3rror** [thonny/thonny#2123]
Hello,

Good day!

I cannot paste, debug, or run the code when I open the app. I don't know what happened, and yesterday it was ok.

I have already uninstalled and reinstalled it, and I also tried to restart the mac. Same result I get.

Thank you for helping me in advance.

<img width="670" alt="Screen Shot 2022-01-15 at 6 07 19 PM" src="https://user-images.githubusercontent.com/97334701/149618055-61512bc9-e2a6-4edc-819a-39a888da6df6.png">

**Comment by aivarannamaa** [comment]
Thank you for the report!

> I cannot paste, debug, or run the code when I open the app

What does it mean precisely? Do you get an error message, are the commands disabled or just nothing happens when you press the button, select menu-item or enter keyboard shortcut?

Does the problem happen with this script only? From the screenshot it looks like syntax coloring isn't working properly. Can you zip the file as it is and post it under this issue?

**Comment by aivarannamaa** [comment]
Please reopen if you have more information to add

**Comment by Cod3rror** [comment]
> Thank you for the report!
> 
> > I cannot paste, debug, or run the code when I open the app
> 
> What does it mean precisely? Do you get an error message, are the commands disabled or just nothing happens when you press the button, select menu-item or enter keyboard shortcut?
> 
> Does the problem happen with this script only? From the screenshot it looks like syntax coloring isn't working properly. Can you zip the file as it is and post it under this issue?

I am sorry for replying just now.

I have no error message. Just nothing happen when I press button(i cannot press button). Also I cannot paste any script.

let me try again first, it might no longer issue now. Actually, thonny helps me alot to understand the code when I am learning python.

**Comment by aivarannamaa** [comment]
If you still have this problem then you could try the first beta of 4.0: https://github.com/thonny/thonny/releases

**Comment by Cod3rror** [comment]
Ok thanks. Ill check it when I got home.

**Comment by Cod3rror** [comment]
> If you still have this problem then you could try the first beta of 4.0: https://github.com/thonny/thonny/releases

I think the code I tried to test when I am a beginner is the only one that makes the Thonny hang. I tried my new and more complex codes and all was working. you could check this line of code and try it. it might be a bug.

#Write your code below this line 👇
def prime_checker(number):
    if number == 2:
        print("It's a prime number.")
    elif number % 2 >= 1:
        print(f"It's a prime number.")
    elif number % 3 >= 1:
        print(f"It's a prime number.")
    else:
        print("It's not a prime number.")


#Write your code above this line 👆
    
#Do NOT change any of the code below👇
n = int(input("Check this number: "))
prime_checker(number=n)

**Comment by Cod3rror** [comment]
I found the bug.. the emoticons.

**Comment by aivarannamaa** [comment]
The 4.0 branch will use the newest Tk, and this is the best I can do about emoticon support.
